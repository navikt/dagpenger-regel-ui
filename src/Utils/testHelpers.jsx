export const noop = () => {};

export const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));


export const formPropsMock = {
  dirty: false,
  errors: {},
  handleBlur: noop, // (eventOrPath),
  handleChange: noop, // (eventOrPath)
  handleReset: noop, // ()
  handleSubmit: noop, // (e)
  initialValues: {},
  isSubmitting: false,
  isValid: false,
  isValidating: false,
  registerField: noop, // (name, Comp)
  resetForm: noop, // (nextValues)
  setError: noop, // (error)
  setErrors: noop, // (errors)
  setFieldError: noop, // (field, message)
  setFieldTouched: noop, // (field, touched, shouldValidate)
  setFieldValue: noop, // (field, value, shouldValidate)
  setFormikState: noop, // (s, callback)
  setStatus: noop, // (status)
  setSubmitting: noop, // (isSubmitting)
  setTouched: noop, // (touched)
  setValues: noop, // (values)
  status: undefined,
  submitCount: 0,
  submitForm: noop, // ()
  touched: {},
  unregisterField: noop, // (name)
  validateField: noop, // (field)
  validateForm: noop, // (values)
  validateOnBlur: true,
  validateOnChange: true,
};
