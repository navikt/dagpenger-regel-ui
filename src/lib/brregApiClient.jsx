import { join } from "path";
import { fetcher } from "./api/fetcher";

const apiUri = join(process.env.BASE_URL || "", "/api/enhet");

export const getOrganisasjonsNavn = async (orgNr) =>
  fetcher(`${apiUri}/${orgNr}`).catch((err) => console.error(err));
