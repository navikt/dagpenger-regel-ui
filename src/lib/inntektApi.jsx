import axios from 'axios';

const apiUri = '/api/v1/inntekt';
const apiUklassifisertUri = `${apiUri}/uklassifisert`;
const apiUncachedUri = `${apiUklassifisertUri}/uncached`;

export const getInntekt = async uri => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUklassifisertUri}/${uri.personId}/${uri.vedtakId}/${uri.beregningsdato}`,
    });
  } catch (error) {
    return error;
  }
};

export const getUncachedInntekt = async uri => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUncachedUri}/${uri.personId}/${uri.vedtakId}/${uri.beregningsdato}`,
    });
  } catch (error) {
    return error;
  }
};

// todo post til uncached hvis henter nye opplysninger
// todo legge til ${request.aktÃ¸rId}/${request.vedtakId}/${request.beregningsdato}`,
export const lagreInntekt = async (request, isUncached, uri) => {
  try {
    return await axios({
      method: 'post',
      url: isUncached
        ? `${apiUncachedUri}/${uri.personId}/${uri.vedtakId}/${uri.beregningsdato}`
        : `${apiUklassifisertUri}/${uri.personId}/${uri.vedtakId}/${uri.beregningsdato}`,
      data: request,
    });
  } catch (error) {
    return error;
  }
};
