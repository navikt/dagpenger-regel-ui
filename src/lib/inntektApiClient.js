import axios from 'axios';

export const getInntekt = async (request) => await axios({
  method: 'post',
  url: `/api/v1/inntekt/uklassifisert`,
  data: request,
}).then(response => response)
  .catch(error => error);

export const lagreInntekt = async (request) => await axios({
  method: 'post',
  url: `/api/v1/inntekt/uklassifisert/update`,
  data: request,
}).then(response => response)
  .catch(error => error);
