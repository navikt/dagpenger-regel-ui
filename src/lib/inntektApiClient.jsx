import axios from 'axios';

const apiUri = '/api/v1/inntekt';
const apiUklassifisertUri = `${apiUri}/uklassifisert`;
const apiUncachedUri = `${apiUklassifisertUri}/uncached`;

export const getInntekt = async (uri) => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUklassifisertUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`,
    });
  } catch (error) {
    return error;
  }
};


export const getUncachedInntekt = async (uri) => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUncachedUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`,
    });
  } catch (error) {
    return error;
  }
};

// todo post til uncached hvis henter nye opplysninger
// todo legge til ${request.aktørId}/${request.vedtakId}/${request.beregningsDato}`,
export const lagreInntekt = async (request, isUncached, uri) => {
  try {
    return await axios({
      method: 'post',
      url: isUncached
        ? `${apiUncachedUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`
        : `${apiUklassifisertUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`,
      data: request,
    });
  } catch (error) {
    return error;
  }
};

export const getVerdikoder = async () => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUri}/verdikoder`,
    });
  } catch (error) {
    return error;
  }
};

