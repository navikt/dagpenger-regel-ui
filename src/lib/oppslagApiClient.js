import axios from 'axios';

const apiUri = '/oppslag/api';

export const  getOrganisasjonsNavn = async (orgNr) => {
  try {
    console.log(`${orgNr}`)
    return await axios({
      method: 'get',
      url: `${apiUri}/organisasjon/${orgNr}`,
    });
  } catch (error) {
    return error;
  }
};


export const getName = async (request) => {
  try {
    return await axios({
      method: 'post',
      url: `${apiUri}/person/name"`,
      data: request,
    });
  } catch (error) {
    return error;
  }
};
