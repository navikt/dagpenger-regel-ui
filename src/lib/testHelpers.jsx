export const noop = () => {};

export const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
