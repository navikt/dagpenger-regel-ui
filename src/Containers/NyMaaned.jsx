import React from 'react';
import PropTypes from 'prop-types';
import { withFormik } from 'formik';
import { Hovedknapp, Knapp } from 'nav-frontend-knapper';
import { Undertittel } from 'nav-frontend-typografi';
import InputField from '../Components/InputField';
import { required } from '../Utils/validering';
import { DisplayFormikState } from '../Utils/formikUtils';

// todo lukke modal onsubmit
const NyMaaned = (props) => {
  const {
    values,
    handleSubmit,
    isSubmitting,
    closeModal,
  } = props;
  return (
    <>
      <Undertittel>
      Legg til ny måned
      </Undertittel>

      <InputField
        name="aarMaaned"
        label="Måned"
        type="month"
        validate={[required]}
      />

      <div className="knapprad">
        <Hovedknapp
          htmlType="submit"
          onClick={() => handleSubmit()}
        >
      Legg til
        </Hovedknapp>
        <Knapp
          htmlType="button"
          onClick={closeModal}
          disabled={isSubmitting}
        >
      Avbryt
        </Knapp>
      </div>

      <DisplayFormikState {...values} />

    </>
  );
};

NyMaaned.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  values: PropTypes.shape().isRequired,
};

export default withFormik({
  mapPropsToValues: ({
    arrayHelpers, closeModal,
  }) => ({
    aarMaaned: '2018-01',
    closeModal,
    arrayHelpers,
  }),
  // Custom sync validation
  validate: (values) => {
    const errors = {};

    return errors;
  },

  handleSubmit: (values, { setSubmitting }) => {
    const { arrayHelpers, closeModal } = values;
    arrayHelpers.push({
      aarMaaned: values.aarMaaned,
      arbeidsInntektInformasjon: {
        inntektListe: [],
      },
    });
    setSubmitting(false);
    closeModal();
  },

  displayName: 'nyMåned',

})(NyMaaned);
