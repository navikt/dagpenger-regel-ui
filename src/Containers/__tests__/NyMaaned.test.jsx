import React from 'react';
import renderer from 'react-test-renderer';
import { Hovedknapp } from 'nav-frontend-knapper';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6
import { noop } from '../../lib/testHelpers';


import NyMaaned from '../NyMaaned';

const shallow = new ShallowRenderer();

const arrayHelpers = {};

const setMånedModal = () => {};
const InitialValues = { name: 'jared' };

test('Skal vise ny skjema for ny måned', () => {
  const component = renderer.create(
    <NyMaaned closeModal={() => setMånedModal(false)} arrayHelpers={arrayHelpers} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

xtest('Skal sende inn ny måned', () => {
  /*
  const component = renderer.create(
    withFormik({
      mapPropsToValues: () => InitialValues,
      handleSubmit: noop,
    })(<NyMaaned closeModal={() => setMånedModal(false)} arrayHelpers={arrayHelpers} />),
  );
  console.log(component.mount);
  // const { root } = component;
  const button = root.findByType(Hovedknapp);

  expect(root.props.submitCount).toBe(0);
  expect(root.props.isSubmitting).toBe(false);
  button.props.onClick();
  expect(root.props.submitCount).toBe(1);
  expect(root.props.isSubmitting).toBe(true);
  */

  const component = shallow.render(
    <NyMaaned closeModal={() => setMånedModal(false)} arrayHelpers={arrayHelpers} />,
  );
  console.log(component.WrappedComponent);
  const button = component.find(Hovedknapp);
  console.log(button);
  expect(component.props.submitCount).toBe(0);
  expect(component.props.isSubmitting).toBe(false);
  button.props.onClick();
  expect(component.props.submitCount).toBe(1);
  expect(component.props.isSubmitting).toBe(true);
});
