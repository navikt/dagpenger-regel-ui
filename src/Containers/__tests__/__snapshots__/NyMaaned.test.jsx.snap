// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skal vise ny skjema for ny måned 1`] = `
Array [
  <h2
    className="typo-undertittel"
  >
    Legg til ny måned
  </h2>,
  <div
    className="skjemaelement"
  >
    <label
      className="skjemaelement__label"
      htmlFor="aarMaaned"
    >
      <div
        className="flex label"
      >
        Måned
      </div>
    </label>
    <input
      autoComplete="off"
      className="skjemaelement__input input--fullbredde"
      id="aarMaaned"
      name="aarMaaned"
      onBlur={[Function]}
      onChange={[Function]}
      readOnly={false}
      type="month"
      value="2018-01"
    />
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <div
    className="knapprad"
  >
    <button
      className="knapp knapp--hoved"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Legg til
    </button>
    <button
      className="knapp knapp--standard"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Avbryt
    </button>
  </div>,
  <div
    style={
      Object {
        "margin": "1rem 0",
      }
    }
  >
    <pre
      style={
        Object {
          "background": "#f6f8fa",
          "fontFamily": "monospace",
          "fontSize": "0.85rem",
          "padding": "0.5rem",
        }
      }
    >
      <strong>
        props
      </strong>
       
      =
       
      {
  "aarMaaned": "2018-01",
  "arrayHelpers": {}
}
    </pre>
  </div>,
]
`;
